Doc/library/functions.rst

```diff
@@ -1689,16 +1689,16 @@ are always available.  They are listed here in alphabetical order.
    With three arguments, return a new type object.  This is essentially a
    dynamic form of the :keyword:`class` statement. The *name* string is the
    class name and becomes the :attr:`~definition.__name__` attribute; the *bases*
-   tuple itemizes the base classes and becomes the :attr:`~class.__bases__`
-   attribute; and the *dict* dictionary is the namespace containing definitions
-   for class body and is copied to a standard dictionary to become the
-   :attr:`~object.__dict__` attribute.  For example, the following two
+   tuple contains the base classes and becomes the :attr:`~class.__bases__`
+   attribute (with the addition of :class:`object`, the ultimate base of all classes);
+   and the *dict* dictionary is the namespace containing attribute and method definitions
+   for the class body.  For example, the following two
    statements create identical :class:`type` objects:
 
       >>> class X:
       ...     a = 1
       ...
-      >>> X = type('X', (object,), dict(a=1))
+      >>> X = type('X', (), dict(a=1))
```

Comment:
While you're changing this, maybe use a dict literal?
```suggestion
      >>> X = type('X', (), {'a': 1})
```

Reply:
I changed it to a literal in my first commit but then decided that I like how with `dict()` you can't accidentally define an attribute that can only be accessed with `getattr`. eg. `{"123": 1}` or even worse, create an object with non-string attributes:

```python
>>> X = type("X", (), {123: 1})
>>> getattr(X(), 123)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: getattr(): attribute name must be string
>>> getattr(X(), "123")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'X' object has no attribute '123'
>>> dir(X())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '<' not supported between instances of 'str' and 'int'
```

because keyword arguments must be [identifiers](https://docs.python.org/3/reference/lexical_analysis.html#identifiers) and also `dict(a=1)` looks really similar to `a = 1`. What do you think, am I overthinking it?

Reply:
I’m +1 to the `dict(a=1)` style here.

Reply:
@taleinat

Reply:
That's a good reason. I'm fine with this.